main.cpp

```diff
@@ -56,46 +56,70 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(int v, const char* text) 
+    : value(v), 
+      name(text) 
+    {
+
+    }  //1
+
+    int value; //2
+    std::string name; //3
 };
 
-struct <#structName1#>                                //4
+struct Comparison                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if (a != nullptr && b != nullptr)
+        {
+            if( a->value < b->value ) return a;
+            if( a->value > b->value ) return b;
+        }
         return nullptr;
     }
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float value1 { 0 }, value2 { 0 };
+    float staticFunc(float* newVal)      //12
     {
-        
+        if (newVal != nullptr)
+        {
+            std::cout << "U's value1 value: " << this->value1 << std::endl;
+            this->value1 = *newVal;
+            std::cout << "U's value1 updated value: " << this->value1 << std::endl;
+            while( std::abs(this->value2 - this->value1) > 0.001f )
+            {
+                this->value2 += this->value1 / this->value2;
+            } 
+            
+            std::cout << "U's staticVal2 updated value: " << this->value2 << std::endl;
+        }
+        return this->value2 * this->value1;
     }
 };
 
-struct <#structname2#>
+struct V
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float staticFunc(U* that, float* newVal )        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        if (that && newVal != nullptr)
```

Comment:
the `&&` operator doesn't work like that.  You need to compare both variables with `nullptr`
```
if( that != nullptr && newVal != nullptr )
```
what you wrote is evaluated as: `if( 'that' is anything other than 0 and 'newVal' is not a nullptr )`
- [ ] revise